### ISSUE TITLE
Devices gone after ptp or reboot of homey

### ISSUE THREAD

**Issue by EDVDVD** [swttt/com.swttt.homekit#131]
Sometimes when homey is rebooted for updates or after a ptp becouse it is offline some homekit devices go missing in the homekit app on Ios.
The devices will return after reboot of the homeykit app on homey, but are placed in the same room as where homey is placed.
So homekit handles it as new devices, also the name is back to the default name in homey of the reapeared devices.

This is really anoying becouse everytime it happens you have to rename and place all the “lost” devices again.

Do you have any idea why this happens?
Only homeykit is installed, the build in support is disabled.

**Comment by robertklep** [comment]
Do you have to re-add the Homey accessory to iOS after a reboot/PTP? Otherwise, I don't have a clue how this can happen (but it has happened to others as well, so it's not a unique problem).

From what I understand of the HomeKit protocol itself, it uniquely identifies devices by a combination (?) of Manufacturer, Model, Name and Serial Number. The data on which the values for these properties is based should remain the same across Homey reboots, so that shouldn't be causing the problem.

**Comment by jahnkenneth** [comment]
This also happens here... After reboot all room connections, names and automations are reset...

**Comment by robertklep** [comment]
Thanks for the info! Might be something I can investigate. I'll see if [Homebridge](https://github.com/nfarina/homebridge) users have similar experiences, because Homebridge and Homeykit are based on the same HomeKit-protocol implementation.

**Comment by jahnkenneth** [comment]
I also use HomeBridge on a Raspberry Pi, and the problem is nonexisting there...

**Comment by robertklep** [comment]
I already found issues on Homebridge's Github where people lose their setups in a similar way, so it happens there as well.

**Comment by robertklep** [comment]
Oh, one additional question: do you have other people in your household that are also using Home(y)Kit?

**Comment by jahnkenneth** [comment]
Ah, Ok. :-) At least it doesnt happen for me on HomeBridge. :-)

**Comment by jahnkenneth** [comment]
Yes, we are multiple users of the Home-app (iPhone) in our household.

**Comment by robertklep** [comment]
And the problem happens for both users?

**Comment by robertklep** [comment]
And another question: which version of Homeykit are you using?

**Comment by jahnkenneth** [comment]
Yes, for all users. I am using the latest version 3.0.3

**Comment by robertklep** [comment]
Worth checking out: https://github.com/nfarina/homebridge/issues/874#issuecomment-277811004

> "I just shut down the iPad (my control center) and rebooted it, which seems to have solved all the problems described"

**Comment by jahnkenneth** [comment]
Hm... Will try this next time it happens... :-) 
But it seems odd that a reboot should bring all the custom settings and automations back...

**Comment by EDVDVD** [comment]
I use homey 2.0.2 and homeykit 3.0.4 but also happened on all versions before that.
As well all homey versions before it.
Also had same issue on the other homekit application of homey and the build in one.
So it is somehow triggered by the way thit implementation works on homey...

Only user on homekit, homepod as hub, phone and 3 ipads as ‘clients’

**Comment by jahnkenneth** [comment]
I agree. This also happens When using the HomeKit app by Sprut1 on Homey...

**Comment by jahnkenneth** [comment]
Some extra info: This does not happen with the Homekit component on HomeAssistant, at least not for me... :-)

**Comment by robertklep** [comment]
Hopefully fixed in v3.0.6 which I just submitted as beta.

**Comment by EDVDVD** [comment]
Will install it as soon as its in the appstore, what has been changed?

**Comment by robertklep** [comment]
Specifically after a reboot, Homeykit will wait at least 2 minutes before starting the Homekit bridge, to allow all devices to be present.

**Comment by EDVDVD** [comment]
A great, sounds logical that the devices are not availeble yet when homeykit starts.
I shall try a reboot and a ptp after the update, but since it does not always happen it can come back later.

**Comment by robertklep** [comment]
We already had a delay in there: the app would check every 30 seconds to see if the total device count had changed, and if so, it would assume that Homey was still booting. Only when the number of devices didn't change in those 30 seconds would it start the bridge.

But @nhassink and I concluded that perhaps 30 seconds was just too short, so we increased the interval to 2 minutes.

**Comment by jahnkenneth** [comment]
I have a lot of devices and apps, and it takes about 5 minutes for all devices to be available. Can this be set as a custom parameter?

**Comment by robertklep** [comment]
Hmm good idea. I think you're still safe, because the 2 minutes is an interval and not a fixed time, but making it configurable sounds like a good idea. I'll create a new issue for that.

**Comment by jahnkenneth** [comment]
😀👍🏻

**Comment by EDVDVD** [comment]
Did 2 reboots and a ptp after update to 3.06, looks like it is solved.
But as it not always happens i will keep you updated if it returns.
But for now thanks for the quick and great support!!!

**Comment by EDVDVD** [comment]
Had it happen again, indeed homeykit starts before all devices are started in homey.
Configurable settle time would be the solution for homey’s with lots of apps/devices.

**Comment by EDVDVD** [comment]
Thanks for the fast response :).
It does take a long time sometimes for everything to load, for me it seems to be always the same devices that get forgotten, xiaomi sensors(zigbee)and google nest protects.
When i open the homey app after reboot those devices are on an red explanation mark for a long time, so it can happen that homey strugles to load both apps/driver and take more than 2 minutes for it, I found out that when i open the homey app at that stage it also needs to feed the app with info so resouces are even more problematic.
The option to start the connection to homekit after 5 minutes of no new devices should solve this.

**Comment by robertklep** [comment]
> When i open the homey app after reboot those devices are on an red explanation mark for a long time

Ah right, I think I know what may be happening: sensor devices, especially if they are running on batteries, are almost always sleeping. Only when they wake up will Homey get notified that they are online, and that make take some time (depending on the sensor).

Homeykit doesn't add devices that are not marked as "ready", like those sensors. If, after some time, the sensors check in with Homey, they are also noticed by Homeykit and added, but that is likely to be _after_ the Homekit bridge has already started. Which means that iOS will think that the device is new.

So I don't think that it's related to Homey being slow, I think that it's because it will take some time for the sensors to wake up and notify Homey that they are alive.

**Comment by jahnkenneth** [comment]
Any way of fixing this? Can HomeyKit add devices that are not marked as ready?

**Comment by robertklep** [comment]
> Can HomeyKit add devices that are not marked as ready?

We added the "ready" check to prevent problems that Homeykit was having with "un-ready" devices, so no :(

But perhaps the settle time configuration is enough for your situation, if you set it appropriately high.

**Comment by robertklep** [comment]
3.0.7 is published as a beta, with a configurable settle time.

**Comment by EDVDVD** [comment]
Great, installed the beta and set on 300 seconds.
Will let you know if problem comes back(probely not) :)
